core/blockchain.go

```diff
@@ -175,6 +175,7 @@ type BlockChain struct {
 	snaps  *snapshot.Tree // Snapshot tree for fast trie leaf access
 	triegc *prque.Prque   // Priority queue mapping block numbers to tries to gc
 	gcproc time.Duration  // Accumulates canonical block processing for trie dumping
+	triedb *trie.Database // The database handler for maintaining trie nodes.
```

Comment:
Can you explain why we are adding this? Previously, we use `bc.stateCache.TrieDB()`, so are they different? If so, then, is the logic changed here?

Reply:
As I understood, it just move out the triedb object from stateCache obj to blockchain obj, behind they still point to same db same struct and config.

Reply:
one reason, maybe they wanna decouple triedb object from cache state for easily feature enhancement 🤔